lib/taskana-core/src/test/java/acceptance/classification/CreateClassificationAccTest.java

```diff
@@ -1,297 +1,374 @@
+        classificationService.newClassification("KeyErrCreation", "", "TASK");
+    classification.setParentKey(invalidParentKey);
+
+    // Make sure that there is no (Parent) Classification with this Key.
+    List<ClassificationSummary> classifications =
+        classificationService.createClassificationQuery().keyIn(invalidParentKey).list();
+    assertThat(classifications).isEmpty();
+    assertThatThrownBy(() -> classificationService.createClassification(classification))
+        .isInstanceOf(InvalidArgumentException.class)
+        .hasMessage("Parent classification could not be found.");
   }
 
   @WithAccessId(user = "businessadmin")
   @Test
-  void testCreateClassificationWithInvalidParentId() {
-    Classification classification = CLASSIFICATION_SERVICE.newClassification("Key5", "", "TASK");
-    classification.setParentId("ID WHICH CANT BE FOUND");
-    assertThatThrownBy(() -> CLASSIFICATION_SERVICE.createClassification(classification))
-        .isInstanceOf(InvalidArgumentException.class);
+  void should_ThrowException_TryingToCreateClassificationWithInvalidParentId() {
+    String invalidParentId = "UNKNOWN_ID";
+
+    Classification classification = classificationService.newClassification("KeyErr", "", "TASK");
+    classification.setParentId(invalidParentId);
+
+    // Make sure that there is no (Parent) Classification with this Id.
+    List<ClassificationSummary> classifications =
+        classificationService.createClassificationQuery().idIn(invalidParentId).list();
+    assertThat(classifications).isEmpty();
+    assertThatThrownBy(() -> classificationService.createClassification(classification))
+        .isInstanceOf(InvalidArgumentException.class)
+        .hasMessage("Parent classification could not be found.");
   }
 
   @WithAccessId(user = "businessadmin")
   @Test
-  void testCreateClassificationWithInvalidParentKey() {
-    Classification classification = CLASSIFICATION_SERVICE.newClassification("Key5", "", "TASK");
-    classification.setParentKey("KEY WHICH CANT BE FOUND");
-    assertThatThrownBy(() -> CLASSIFICATION_SERVICE.createClassification(classification))
-        .isInstanceOf(InvalidArgumentException.class);
+  void should_ThrowException_TryingToCreateClassificationWithExplicitId() {
+    String explicitId = "EXPLICIT ID";
+
+    ClassificationImpl classification =
+        (ClassificationImpl) classificationService.newClassification("KeyErrCreation", "", "TASK");
+    classification.setId(explicitId);
+
+    assertThatThrownBy(() -> classificationService.createClassification(classification))
+        .isInstanceOf(InvalidArgumentException.class)
+        .hasMessage("ClassificationId should be null on creation");
+  }
+
+  @WithAccessId(user = "taskadmin")
+  @WithAccessId(user = "user-1-1")
+  @TestTemplate
+  void should_ThrowException_When_UserRoleIsNotAdminOrBusinessAdmin() {
+    Classification classification =
+        classificationService.newClassification("KeyErrCreation", "", "TASK");
```

Comment:
Is there a reason you didn't increment the Key for the error tests?

Reply:
Yes, I used this Key for tests which shouldn't create any classification because of exceptions etc. The main reason for this was the "should_NotCreateClassification_When_ExceptionIsThrownDuringCreation()" test (@AfterAll) I implemented. I had the urge to test if there was no classification created by mistake :)) So, I thought using the same error key would save some code. Further, I think it look cleaner :D